src/plugins/intel_cpu/tests/functional/single_layer_tests/select.cpp

```diff
@@ -1,174 +1,206 @@
+#include "test_utils/fusing_test_utils.hpp"
+#include "ngraph_functions/builders.hpp"
+
+using namespace InferenceEngine;
+using namespace CPUTestUtils;
+using namespace ov::test;
+
+namespace CPULayerTestsDefinitions {
+
+using selectParams = std::tuple<std::vector<InputShape>,         // input shapes
+                                ElementType,                     // Then/Else precision
+                                ngraph::op::AutoBroadcastSpec,   // broadcast
+                                fusingSpecificParams>;
+
+class SelectLayerCPUTest : public testing::WithParamInterface<selectParams>,
+                           virtual public SubgraphBaseTest,
+                           public CpuTestWithFusing {
+public:
+   static std::string getTestCaseName(testing::TestParamInfo<selectParams> obj) {
+       std::vector<InputShape> shapes;
+       ElementType precision;
+       ngraph::op::AutoBroadcastSpec broadcast;
+       fusingSpecificParams fusingParams;
+       std::tie(shapes, precision, broadcast, fusingParams) = obj.param;
+
+       std::ostringstream result;
+       result << "Condition_prc_" << ElementType::boolean << "_Then_Else_prc_" << precision << "_";
+       result << "IS=(";
+       for (const auto& shape : shapes) {
+           result << shape.first << "_";
+       }
+       result << ")_TS=(";
+       for (const auto& shape : shapes) {
+           for (const auto& item : shape.second) {
+               result << CommonTestUtils::vec2str(item) << "_";
+           }
+       }
+       result << "Broadcast=" << broadcast.m_type;
+       result << CpuTestWithFusing::getTestCaseName(fusingParams);
+
+       return result.str();
+   }
+
+protected:
+   void SetUp() override {
+        abs_threshold = 0;
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+        std::vector<InputShape> shapes;
+        ElementType precision;
+        ngraph::op::AutoBroadcastSpec broadcast;
+        fusingSpecificParams fusingParams;
+        std::tie(shapes, precision, broadcast, fusingParams) = this->GetParam();
+        init_input_shapes(shapes);
+        std::tie(inFmts, outFmts, priority, selectedType) = emptyCPUSpec;
+        selectedType = makeSelectedTypeStr(getPrimitiveType(), ov::element::i8);
+
+        auto parameters = ngraph::builder::makeDynamicParams(ov::element::TypeVector{ov::element::boolean, precision, precision}, inputDynamicShapes);
+        auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(parameters));
+        auto select = ngraph::builder::makeSelect(paramOuts, broadcast);
+
+        function = makeNgraphFunction(precision, parameters, select, "Eltwise");
+   }
+
+   void generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+
+        auto cond_tensor = utils::create_and_fill_tensor(funcInputs[0].get_element_type(), targetInputStaticShapes[0], 1, 0, 1);
```

Comment:
To be honest I'm not sure that the tensor contains the both `true` and `false` here. I think there are only `false` values. By the way, I have locally checked values of tensor and I saw really only `false` values.

So I'd suggest to use the same [generation](https://github.com/openvinotoolkit/openvino/blob/bc663878eb5ecf2afb7a982ec85918285bc9c5b6/src/tests/functional/plugin/shared/src/snippets/select.cpp#L16) as the `Select` in `Snippets`. It generates different values. Also I'd suggest to generate different values for `then` and `else` inputs to see that `Select` correctly chooses values.

Reply:
My bad. Thanks a lot for noticing this defect! Corrected